addons/web/controllers/main.py

```diff
@@ -1445,6 +1445,11 @@ def _content_image(self, xmlid=None, model='ir.attachment', id=None, field='data
         if status in [301, 304] or (status != 200 and download):
             return request.env['ir.http']._response_by_status(status, headers, image_base64)
         if not image_base64:
+            if placeholder == 'placeholder.png' and model in request.env:
+                placeholder = request.env[model]._get_placeholder_filename()
+                status = 200
+            if model == 'res.users' and id == '-1':
+                placeholder = 'user-slash.png'
```

Comment:
If ``_get_placeholder_filename`` is overwritten on the ``res.users`` model, it would be handled for this specific use case.